src/market/directory.rs

```diff
@@ -251,11 +262,20 @@ fn handle_client(stream: &mut TcpStream, addresses: Arc<RwLock<HashSet<String>>>
 
     if request_line.starts_with("POST") {
         let addr: String = request_line.replace("POST ", "").trim().to_string();
-        addresses.write().unwrap().insert(addr.clone());
+        directory.addresses.write().unwrap().insert(addr.clone());
         log::info!("Got new maker address: {}", addr);
+
+        let address_file = directory.data_dir.join("addresses.dat");
+        let mut file = OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(address_file)
+            .unwrap();
+        writeln!(file, "{}", addr).unwrap();
```

Comment:
I'm unsure about this approach, as it involves performing an expensive I/O operation with every POST request. A better solution might be to implement the `Drop` trait for the Directory Server object, so when the object is dropped, the addresses are written to the file. This would reduce the thread load caused by frequent file I/O operations. To handle abrupt terminations, we can use signal handling to catch such instances. For scenarios involving panics, periodic writes could be employed. The goal of these optimizations is to avoid burdening the server thread with heavy I/O operations. 

However, on second thought, the Directory Server might not face significant load during the initial stages, so this approach should work for now.

Reply:
Shall I put a comment for optimizing it later, and work on closing https://github.com/citadel-tech/coinswap/pull/243?

Reply:
That works, just add a comment for potential improvement. And squash the commits. We are green with this.

Reply:
@claddyy, could you remove those `unwraps` and introduce specific error types for these cases?

Reply:
@Shourya742 https://github.com/citadel-tech/coinswap/pull/269/files#diff-4c0f7e79b7d4fa2d8b480dfda773b540f40d9cb26a9c0955073ffa5b548f9962R198
nit: Shall I update this to `address_arc`?

Reply:
Sure!!! nice catch,..

Reply:
> @claddyy, could you remove those `unwraps` and introduce specific error types for these cases?

addressed https://github.com/citadel-tech/coinswap/commit/bd37c5f334d242599bee95988092a0c437285a1f